code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1336,4 +1336,151 @@
+	glass_desc = "Looks like some chunky dark red mix."
+	reagent_weight = 4
+	addiction_threshold = 40
+	// Max healing non OD with best rolls + buffs: 0.2 + 0.15 + 0.2 = 0.55 
+	// Max healing with OD + best rolls and all buffs: 0.4 + 0.35 + 0.4 = 1.15
+
+/datum/reagent/medicine/burnmix/on_mob_life(mob/living/carbon/M)
+	var/heal_roll = pick(-0.1,-0.2) // Base heal, Non OD
+	var/slur = 10 // Slur 10 sets up for future slur gains as real effects seem to start at 30
+	var/jitter = 0
+	if(M.health <= M.crit_threshold)
+		heal_roll = 0 // Burnmix will not heal anyone inside of crit without epipen or morhpine
+	if(prob(10))
+		heal_roll = 0 // Burnmix has a chance not to basic heal at all
+	if(holder.has_reagent(/datum/reagent/medicine/epinephrine)) // start IF has Epinephrine in body
+		jitter = jitter + 5
+		slur = slur + 25
+		heal_roll = heal_roll - 0.15
+		if(prob(15))
+			M.losebreath++
+			if(prob(80))
+				M.adjustStaminaLoss(1.25*REM, 0) // end IF has Epinephrine in body
+
+	if(holder.has_reagent(/datum/reagent/medicine/perfluorodecalin)) // start IF has Perfluorodecalin in body
+		jitter = jitter + 5
+		slur = slur + 25
+		M.drowsyness += 0.5
+		heal_roll = heal_roll - 0.2 // start IF has Perfluorodecalin in body
+
+	if(prob(10)) // Start of base probability effects
+		var/tox_roll = pick(1,-1) // Rolls for damage or healing
+		if(tox_roll == 1)
+			M.emote("moan")
+		if(tox_roll == -1)
+			M.emote("twitch")
+		if(prob(90))
+			M.adjustToxLoss(tox_roll*REM, 0)
+		M.Jitter(jitter)
+	if(prob(1)) // Last set of non OD probability effects
+		M.Dizzy(jitter)
+		M.slurring = slur // End of base probability effects
+	if(heal_roll < 0) // Healing payload after calculations
+		if (M.stat == DEAD) // Non OD , if dead , 30% chance to discard healing payload
+			if(prob(70))
+				M.adjustFireLoss(heal_roll*REM, FALSE, FALSE, BODYPART_ORGANIC)
+				M.adjustBruteLoss(heal_roll*REM, 0)
+		else
+			M.adjustFireLoss(heal_roll*REM, 0) 
+			M.adjustBruteLoss(heal_roll*REM, 0) 
+	..()
+	. = 1
+
+/datum/reagent/drug/burnmix/overdose_start(mob/living/M)
+	to_chat(M, "<span class='userdanger'>You feel the BurnMix increase in potency.</span>")
+
+
+/datum/reagent/medicine/burnmix/overdose_process(mob/living/M)
+	var/heal_roll = pick(-0.2,-0.4)
+	var/slur = 20
+	var/jitter = 2
+	if(M.health <= M.crit_threshold)
+		heal_roll = heal_roll + 0.2 // Burnmix in overdose will heal inside of crit without epipen or perfluorodecalin. Being in crit means a roll for either no base healing or half reduction to base heal.
+	if(heal_roll > 0 && prob(10))
+		heal_roll = 0 // Burnmix has a chance not to basic heal at all
+	if(holder.has_reagent(/datum/reagent/medicine/epinephrine))
+		heal_roll = heal_roll - 0.35
+		jitter = jitter + 15
+		slur = slur + 25
+		if(prob(25))
+			M.losebreath++
+			M.adjustStaminaLoss(2.5*REM, 0)
+			if(prob(80))
+				M.adjustToxLoss(0.5*REM, 0)
+				M.adjustToxLoss(pick(0,-0.25), 0) // gives a chance to negate half epinephrine tox damage
```

Comment:
Instead of this thing do `M.adjustToxLoss(pick(0.25, 0.5)*REM, 0)`

Reply:
I'll do this when I get home. 
Thank you!

Reply:
Now that I'm home and had a chance to re-look at my code I can say that this is intentional.
Why I did this:
When rolling a 0 there is a chance to heal 0.25 tox damage per ticket. I want to be able to heal the player if good RNG happens. 

That being said it looks like the same result can be accomplished with a weighted pick statement so I am going to go ahead and re-write that portion if number of lines is a concearn.

Reply:
Resolved. Changed both lines to to fit inside of  M.adjustToxLoss(pick(0.5,0.25,0,0,-0.25)*REM, 0)